internal/sync/cache.go

```diff
@@ -0,0 +1,362 @@
+	"github.com/argoproj/gitops-engine/pkg/health"
+	"github.com/argoproj/gitops-engine/pkg/utils/kube"
+	"github.com/numaproj-labs/numaplane/api/v1alpha1"
+	"golang.org/x/sync/semaphore"
+	v1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+)
+
+// GitOps engine cluster cache tuning options
+var (
+	// clusterCacheResyncDuration controls the duration of cluster cache refresh.
+	// NOTE: this differs from gitops-engine default of 24h
+	clusterCacheResyncDuration = 12 * time.Hour
+
+	// clusterCacheWatchResyncDuration controls the maximum duration that group/kind watches are allowed to run
+	// for before relisting & restarting the watch
+	clusterCacheWatchResyncDuration = 10 * time.Minute
+
+	// clusterSyncRetryTimeoutDuration controls the sync retry duration when cluster sync error happens
+	clusterSyncRetryTimeoutDuration = 10 * time.Second
+
+	// The default limit of 50 is chosen based on experiments.
+	clusterCacheListSemaphoreSize int64 = 50
+
+	// clusterCacheListPageSize is the page size when performing K8s list requests.
+	// 500 is equal to kubectl's size
+	clusterCacheListPageSize int64 = 500
+
+	// clusterCacheListPageBufferSize is the number of pages to buffer when performing K8s list requests
+	clusterCacheListPageBufferSize int32 = 1
+
+	// clusterCacheRetryLimit sets a retry limit for failed requests during cluster cache sync
+	// If set to 1, retries are disabled.
+	clusterCacheAttemptLimit int32 = 1
+
+	// clusterCacheRetryUseBackoff specifies whether to use a backoff strategy on cluster cache sync, if retry is enabled
+	clusterCacheRetryUseBackoff bool = false
+)
+
+type ResourceInfo struct {
+	GitSyncName string
+
+	Health       *health.HealthStatus
+	manifestHash string
+}
+
+type LiveStateCache interface {
+	// GetManagedLiveObjs Returns state of live nodes which correspond for target nodes of specified gitsync.
+	GetManagedLiveObjs(gitsync *v1alpha1.GitSync, targetObjs []*unstructured.Unstructured) (map[kube.ResourceKey]*unstructured.Unstructured, error)
+	// Init must be executed before cache can be used
+	Init() error
+}
+
+type cacheSettings struct {
+	clusterSettings clustercache.Settings
+
+	// ignoreResourceUpdates is a flag to enable resource-ignore rules.
+	ignoreResourceUpdatesEnabled bool
+}
+
+type ObjectUpdatedHandler = func(managedByGitSync map[string]bool, ref v1.ObjectReference)
+
+type liveStateCache struct {
+	clusterCacheConfig *rest.Config
+	logger             *zap.SugaredLogger
+	onObjectUpdated    ObjectUpdatedHandler
+
+	cluster       clustercache.ClusterCache
+	cacheSettings cacheSettings
+	lock          sync.RWMutex
+}
+
+func NewLiveStateCache(
+	onObjectUpdated ObjectUpdatedHandler,
+) LiveStateCache {
+	return &liveStateCache{
+		onObjectUpdated: onObjectUpdated,
+	}
+}
+
+func (c *liveStateCache) getCluster() (clustercache.ClusterCache, error) {
```

Comment:
```suggestion
func (c *liveStateCache) getCache() (clustercache.ClusterCache, error) {
```

Reply:
keeping the name matching the previous comment.

Reply:
sure